flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java

```diff
@@ -294,6 +294,44 @@ default BindingValidationStatus<TARGET> validate() {
          *         changes, otherwise {@literal false}.
          */
         boolean hasChanges();
+
+        /**
+         * Checks whether this binding should be processed during validation and
+         * writing to bean.
+         *
+         * @return {@literal true} if this binding should be processed,
+         *         {@literal false} if this binding should be ignored
+         */
+        default boolean isApplied() {
+            return getIsAppliedPredicate().test(this);
+        }
+
+        /**
+         * Gets predicate for testing {@link #isApplied()}. By default,
+         * non-visible components are ignored during validation and bean
+         * writing.
+         *
+         * @return predicate for testing {@link #isApplied()}
+         */
+        default SerializablePredicate<Binding<BEAN, TARGET>> getIsAppliedPredicate() {
+            return binding -> {
+                if (binding.getField() instanceof Component) {
+                    return ((Component) binding.getField()).isVisible();
```

Comment:
nit: could use pattern matching to simplify the cast operation (unless we plan to back-port the feature)
```suggestion
                if (binding.getField() instanceof Component component) {
                    return component.isVisible();
```

Reply:
Fixed